# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1281293389 14400
#      Sun Aug 08 14:49:49 2010 -0400
# Node ID 045558999570cb8c4e888c93baf5162e86963c8f
# Parent  f065b24f792aaebe4322cb640ab758130285bf69
print.m: Include 'colormap' when converting RGB to mono.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-08-08  Ben Abbott <bpabbott@mac.com>
+
+	* plot/print.m: Include 'colormap' when converting RGB to mono.
+
 2010-08-07  Ben Abbott <bpabbott@mac.com>
 
 	* plot/__fltk_print__.m: When running pstoedit, crop eps file after it
 	is created. Correctly check error status.
 
 2010-08-06  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/cell2mat.m: Allow cells of struct arrays.
diff --git a/scripts/plot/print.m b/scripts/plot/print.m
--- a/scripts/plot/print.m
+++ b/scripts/plot/print.m
@@ -287,31 +287,33 @@ function print (varargin)
         linestyle = "-";
       else
         linestyle = "--";
       endif
       set (h, "linestyle", linestyle)
     endif
 
     if (opts.use_color < 0)
-      color_props = {"color", "facecolor", "edgecolor"};
+      color_props = {"color", "facecolor", "edgecolor", "colormap"};
       for c = 1:numel(color_props)
         h = findobj (opts.figure, "-property", color_props{c});
         hnone = findobj (opts.figure, color_props{c}, "none");
         h = setdiff (h, hnone);
         m = numel (props);
         for n = 1:numel(h)
           rgb = get (h(n), color_props{c});
           props(m+n).h = h(n);
           props(m+n).name = color_props{c};
           props(m+n).value = {get(h(n), color_props{c})};
           xfer = repmat ([0.30, 0.59, 0.11], size (rgb, 1), 1);
-          ## convert RGB color to RGB gray scale
-          ggg = repmat (sum (xfer .* rgb, 2), 1, 3);
-          set (h(n), color_props{c}, ggg)
+          if (isnumeric (rgb))
+            ## convert RGB color to RGB gray scale
+            ggg = repmat (sum (xfer .* rgb, 2), 1, 3);
+            set (h(n), color_props{c}, ggg)
+          endif
         endfor
       endfor
     endif
 
     if (! isempty (opts.font) || ! isempty (opts.fontsize))
       h = findobj (opts.figure, "-property", "fontname");
       m = numel (props);
       for n = 1:numel(h)
